setup.py

```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
+            cuda.cudaGetDeviceCount(ctypes.byref(device_count))
+            if device_count.value > 0:
+                gpu_platform = True
+        except Exception as e:
+            gpu_platform = False
+    if gpu_platform:
+        req = ['mxnet-cu90mkl>=1.2'] + requirements
+    else:
+        req = ['mxnet-mkl>=1.2'] + requirements
+
+    return req
+
+
+class BuildFrontEnd(Command):
+    description = 'Build Model Server Frontend'
+    user_options = []
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        try:
+            front_end_bin_dir = os.path.abspath('.') + '/mms/frontend'
+            os.mkdir(front_end_bin_dir)
+        except OSError as exc:
+            if exc.errno == errno.EEXIST and os.path.isdir(front_end_bin_dir):
+                pass
+            else:
+                raise
+            if os.path.exists(source_server_file):
+                os.remove(source_server_file)
+
+            cwd = os.getcwd()
+            os.chdir(os.path.abspath('./frontend/'))
+            try:
+                subprocess.check_call('./gradlew build', shell=True)
+            except OSError:
+                assert 0, "build failed"
+            os.chdir(cwd)
+            copyfile(source_server_file, dest_file_name)
+
+
+class BuildPy(setuptools.command.build_py.build_py):
+    @staticmethod
+    def sanity_check_success():
+        dirname = os.path.dirname(os.path.abspath(__file__)) + '/frontend/server/build/libs'
+        if os.path.isdir(dirname) and any(".jar" in file_name for file_name in os.listdir(dirname)) and \
+                time.time() - os.stat(source_server_file).st_mtime < 300 and os.path.isfile(dest_file_name):
+            return True
+        return False
+
+    def run(self):
+        if BuildPy.sanity_check_success() is False and 'build_frontend' not in opt_set:
+            sys.stderr.write('\033[91m' +
+                             'Runing \"python setup.py build_frontend\" - missing or old binary found.' +
+                             '\033[0m' + '\n')
+            sys.stderr.flush()
+            self.run_command('build_frontend')
+            setuptools.command.build_py.build_py.run(self)
+
+
+if __name__ == '__main__':
+    opt_set = set(sys.argv)
+    version = detect_model_server_version()
+
+    requirements = ['Pillow', 'psutil', 'importlib2', 'fasteners', 'future; python_version < "3.*"']
+    requirements = select_mxnet(requirements)
+
+    setup(
+        name='mxnet-model-server',
+        version=version.strip()+'a2',
+        description='Model Server for Apache MXNet is a tool for serving neural net models for inference',
+        long_description=pypi_description(),
+        url='https://github.com/awslabs/mxnet-model-server',
+        keywords='MXNet Model Server Serving Deep Learning Inference AI',
+        packages=pkgs,
+        cmdclass={
+            'build_frontend': BuildFrontEnd,
+            'build_py':BuildPy,
```

Comment:
Missing space

Reply:
True added it. I should probably look into attaching a pylint for this file as well.